sdk/core/Azure.Core.Experimental/tests/JsonDataTests.cs

```diff
@@ -23,14 +23,15 @@ public void DefaultConstructorMakesEmptyObject()
         [Test]
         public void CanCreateFromJson()
         {
-            var jsonData = JsonData.FromString("\"string\"");
+            var jsonData = new JsonData("\"string\"");
 
             Assert.AreEqual("\"string\"", jsonData.ToJsonString());
         }
 
         [Test]
         public void CanCreateFromNull()
```

Comment:
@ellismg, what scenario was driving this requirement?

Reply:
Can you clarify what scenario?   I'm not sure if we're discussing creating an instance form a primitive string or from JSON in string form.

Reply:
The scenario of creating a JsonData instance from null -- it creates an ambiguous call site.

Reply:
I believe that was related to the mutable starting point.  There's no benefit to doing so that I can see if our goal is to enable consuming a blob of JSON data.

Reply:
Would it be possible to get a null response.Content in a service response?

Reply:
Maybe?   I'm not sure what we return for content in the case of something like a 204.

Reply:
Yeah, and I guess so long as the null is "typed", it wouldn't result in an ambiguous call site.

Reply:
> @ellismg, what scenario was driving this requirement?

I do not recall a specific scenario driving this requirement (e.g. some service that had some response of this shape.

I think the goal here was we were trying to provide a mapping between CLR and JSON types and "null" is a type in both domains.

(from node)

```
> JSON.parse('null');
null
> JSON.stringify(null)
'null'
>
```

In the same way we have `new JsonData(5)` and `new JsonData("foo")`, I think we wanted to make sure that `new JsonData(null)` (which was going to be valid code since we already had a constructor that took a reference type) did what you "might expect" which is get a JSON null.

I suspect you could change the behavior and the world wouldn't fall down (I feel like if there was a really subtle reason we had to do this I would have written a load bearing comment somewhere or would have at least remembered the rationale better than I do now)